test/test_tensor_creation_ops.py

```diff
@@ -940,9 +941,13 @@ def test_linspace(self, device, dtype):
                          atol=0, rtol=0)
 
         # Check for race condition (correctness when applied on a large tensor).
-        if dtype not in (torch.int8, torch.uint8, torch.int16, torch.half):
-            y = torch.linspace(0, 1000000 - 1, 1000000, device=device, dtype=dtype)
-            cond = y[:-1] < y[1:]
+        if dtype not in (torch.int8, torch.uint8, torch.int16, torch.half, torch.bfloat16):
+            y = torch.linspace(0, 999999 + (999999j if dtype.is_complex else 0),
```

Comment:
I think we should also assert a runtime error for the case where start and end are complex, but the dtype is not specified (and add a note that it needs to be fixed).
also what happens if start and end are complex and dtype=float?

Reply:
It would throw `RuntimeError`, similar to `tensor(1j, dtype=torch.float)`:

    RuntimeError: value cannot be converted to type float without overflow: (0,1)